### PR TITLE
DOP-3278: changes draggable from cell to row

### PR DESCRIPTION
**Pull request by mmeigs**
### Ticket

[DOP-3278](https://jira.mongodb.org/browse/DOP-3278)

### Notes

From Allison Mui's [notes:](https://docs.google.com/document/d/19bFoAk5c8JCkkrs6Rb1zhJR2QjfJsL-a2ZrY-tlMmjY/edit)

The entire row of a draggable table should be draggable, rather than the current iteration which gives only that functionality to the "drag" menu burger icon on the left.

The cursor should be a pointer until it's dragging, which then can become a grabbed hand. Overrode react-beautiful-dnd cursor styles with a more specific css selector.

### CODE REVIEW

**Comment by carooobregon** [comment]
Functionality works well! Just wondering if we also need to eliminate `...rest `from the `Cell` component too?

File: frontend/tests/unit/Repo.test.tsx [hunk 1]
```diff
@@ -55,51 +55,95 @@ describe('Repo', () => {
...
-      code: 'ArrowDown',
-      keyCode: 40,
-      charCode: 40
+      const rowsAfter = await wrapper.findAllByRole('button', { name: /Menu Icon/ });
+      const rowTextsAfter = rowsAfter.map((x: HTMLElement) => x.textContent);
+      expect(rowTextsAfter[1]).toContain(versions[0].gitBranchName);
     });
 
-    expect(await screen.findByText(/You have moved the item/i)).toBeInTheDocument();
+    // const dragButtons = screen.getAllByRole('button', { name: 'Menu Icon' });
```
Comment (rayangler) [inline frontend/tests/unit/Repo.test.tsx:108]: Are the commented lines of code intentionally being kept? If so, we should include a comment explaining why

File: frontend/tests/unit/Repo.test.tsx [hunk 2]
```diff
@@ -55,51 +55,95 @@ describe('Repo', () => {
...
-    render(<Repo repoName="docs" />);
 
-    const rowsBefore = await screen.findAllByRole('row');
-    const rowTextsBefore = rowsBefore.map((x) => x.textContent);
-    expect(rowTextsBefore[1]).toContain(versions[0].gitBranchName);
+    let wrapper: any;
+    await waitFor(() => {
+      wrapper = render(<Repo repoName="docs" />);
+    }).then(async () => {
+      const table = wrapper.getAllByRole('table');
```
Comment (rayangler) [inline frontend/tests/unit/Repo.test.tsx:63]: `table` is unused. Should we remove it?

File: frontend/tests/unit/Repo.test.tsx [hunk 3]
```diff
@@ -55,51 +55,95 @@ describe('Repo', () => {
 describe('Drag and Drop', () => {
   it('moves a row up', async () => {
     const versions = getVersions('docs').branches;
-    render(<Repo repoName="docs" />);
 
-    const rowsBefore = await screen.findAllByRole('row');
```
Comment (rayangler) [inline frontend/tests/unit/Repo.test.tsx:60]: We should remove `screen` as an import since it's now unused

**Review by rayangler — COMMENTED** [review]
Functionality looks great when tested locally! Flagging some things for general cleanup

File: frontend/tests/unit/Repo.test.tsx (hunk 1 above)
Comment (mmeigs) [reply frontend/tests/unit/Repo.test.tsx:108]: Thank you for all these clean-ups!

**Pull request merged**